tfhe/src/integer/ciphertext/compressed_modulus_switched_ciphertext.rs

```diff
@@ -56,5 +68,54 @@ pub type CompressedModulusSwitchedRadixCiphertext =
 ///
 /// assert_eq!(clear, dec);
 /// ```
-pub type CompressedModulusSwitchedSignedRadixCiphertext =
-    BaseSignedRadixCiphertext<CompressedModulusSwitchedCiphertext>;
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct CompressedModulusSwitchedSignedRadixCiphertext(
+    pub(crate) CompressedModulusSwitchedRadixCiphertextGeneric,
+);
+
+impl ParameterSetConformant for CompressedModulusSwitchedSignedRadixCiphertext {
+    type ParameterSet = RadixCiphertextConformanceParams;
+
+    fn is_conformant(&self, params: &RadixCiphertextConformanceParams) -> bool {
+        self.0.is_conformant(params)
+    }
+}
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub(crate) struct CompressedModulusSwitchedRadixCiphertextGeneric {
+    pub paired_blocks: Vec<CompressedModulusSwitchedCiphertext>,
+    pub last_block: Option<CompressedModulusSwitchedCiphertext>,
+}
+
+impl ParameterSetConformant for CompressedModulusSwitchedRadixCiphertextGeneric {
+    type ParameterSet = RadixCiphertextConformanceParams;
+
+    fn is_conformant(&self, params: &RadixCiphertextConformanceParams) -> bool {
+        let mut shortint_params = params.shortint_params;
+
+        shortint_params.degree = Degree::new(
+            MaxDegree::from_msg_carry_modulus(
+                shortint_params.message_modulus,
+                shortint_params.carry_modulus,
+            )
+            .get(),
```

Comment:
In the `Ciphertext` compliance, we expect the degree to be equal to the expected degree, not smaller or equal
It can prevent a user from leaking information, or make results invalid (if the given degree is smaller than the real one).
Ideally, it should be handled with the ZK (if used).